configs/FLYWOOH743PRO/config.h

```diff
@@ -0,0 +1,150 @@
+#define USE_GYRO_SPI_ICM42688P
+#define USE_BARO
+#define USE_BARO_BMP280
+#define USE_BARO_DPS310
+#define USE_MAX7456
+#define USE_SDCARD
+
+#define BEEPER_PIN          PA15
+#define MOTOR1_PIN          PB0
+#define MOTOR2_PIN          PB1
+#define MOTOR3_PIN          PA0
+#define MOTOR4_PIN          PA1
+#define MOTOR5_PIN          PA2
+#define MOTOR6_PIN          PA3
+#define MOTOR7_PIN          PD12
+#define MOTOR8_PIN          PD13
+#define MOTOR9_PIN          PD14
+#define MOTOR10_PIN         PD15
+#define SERVO1_PIN          PE5
+#define SERVO2_PIN          PE6
+#define RX_PPM_PIN          PC7
+#define LED_STRIP_PIN       PA8
+#define UART1_TX_PIN        PA9
+#define UART2_TX_PIN        PD5
+#define UART3_TX_PIN        PD8
+#define UART4_TX_PIN        PB9
+#define UART6_TX_PIN        PC6
+#define UART7_TX_PIN        PE8
+#define UART8_TX_PIN        PE1
+#define UART1_RX_PIN        PA10
+#define UART2_RX_PIN        PD6
+#define UART3_RX_PIN        PD9
+#define UART4_RX_PIN        PB8
+#define UART6_RX_PIN        PC7
+#define UART7_RX_PIN        PE7
+#define UART8_RX_PIN        PE0
+#define I2C1_SCL_PIN        PB6
+#define I2C2_SCL_PIN        PB10
+#define I2C1_SDA_PIN        PB7
+#define I2C2_SDA_PIN        PB11
+#define LED0_PIN            PE3
+#define LED1_PIN            PE4
+#define SPI1_SCK_PIN        PA5
+#define SPI2_SCK_PIN        PB13
+#define SPI3_SCK_PIN        PB3
+#define SPI4_SCK_PIN        PE12
+#define SPI1_SDI_PIN        PA6
+#define SPI2_SDI_PIN        PB14
+#define SPI3_SDI_PIN        PB4
+#define SPI4_SDI_PIN        PE13
+#define SPI1_SDO_PIN        PD7
+#define SPI2_SDO_PIN        PB15
+#define SPI3_SDO_PIN        PB5
+#define SPI4_SDO_PIN        PE14
+#define ADC_VBAT_PIN        PC0
+#define ADC_RSSI_PIN        PC5
+#define ADC_CURR_PIN        PC1
+#define SDIO_CK_PIN         PC12
+#define SDIO_CMD_PIN        PD2
+#define SDIO_D0_PIN         PC8
+#define SDIO_D1_PIN         PC9
+#define SDIO_D2_PIN         PC10
+#define SDIO_D3_PIN         PC11
+#define PINIO1_PIN          PD10 //9V BEC
+#define PINIO2_PIN          PD11 //CAM1,2
+#define MAX7456_SPI_CS_PIN  PB12
+#define GYRO_1_EXTI_PIN     PB2
+#define GYRO_2_EXTI_PIN     PE15
+#define GYRO_1_CS_PIN       PC15
+#define GYRO_2_CS_PIN       PE11
+
+#define TIMER_PIN_MAPPING \
+    TIMER_PIN_MAP( 0, MOTOR1_PIN ,   2,   0) \
+    TIMER_PIN_MAP( 1, MOTOR2_PIN ,   2,   0) \
+    TIMER_PIN_MAP( 2, MOTOR3_PIN ,   2,   0) \
+    TIMER_PIN_MAP( 3, MOTOR4_PIN ,   2,   0) \
+    TIMER_PIN_MAP( 4, MOTOR5_PIN ,   2,   0) \
+    TIMER_PIN_MAP( 5, MOTOR6_PIN ,   2,   0) \
+    TIMER_PIN_MAP( 6, MOTOR7_PIN ,   1,   0) \
+    TIMER_PIN_MAP( 7, MOTOR8_PIN ,   1,   0) \
+    TIMER_PIN_MAP( 8, MOTOR9_PIN ,   1,   0) \
+    TIMER_PIN_MAP( 9, MOTOR10_PIN,   1,   0) \
+    TIMER_PIN_MAP(10, SERVO1_PIN ,   1,  -1) \
+    TIMER_PIN_MAP(11, SERVO2_PIN ,   1,  -1) \
+    TIMER_PIN_MAP(12, LED_STRIP_PIN, 1,   0) \
+    TIMER_PIN_MAP(13, RX_PPM_PIN ,   2,  -1)
+
+#define ADC1_DMA_OPT        8
```

Comment:
do these need defining with the DMA controller?

Reply:
Looks like this is just copied over from another target. I did that as well for a H743 config but turned out it was wrong.

Reply:
Looks like this is just copied over from another target. I did that as well for a H743 config but turned out it was wrong.

Reply:
This is very common setting:

```
src/config/configs/SPRACINGH7EF/config.h:162:#define ADC3_DMA_OPT                  10
src/config/configs/FOXEERH743V2/config.h:107:#define ADC1_DMA_OPT 8
src/config/configs/FOXEERH743V2/config.h:108:#define ADC2_DMA_OPT 9
src/config/configs/FOXEERH743V2/config.h:109:#define ADC3_DMA_OPT 10
src/config/configs/IFLIGHT_H743_AIO/config.h:92:#define ADC1_DMA_OPT 8
src/config/configs/IFLIGHT_H743_AIO/config.h:93:#define ADC3_DMA_OPT 9
src/config/configs/SPRACINGH7EVO/config.h:122:#define ADC3_DMA_OPT                  10
src/config/configs/MINI_H743_HD/config.h:119:#define ADC1_DMA_OPT        8
src/config/configs/MINI_H743_HD/config.h:120:#define ADC2_DMA_OPT        9
src/config/configs/MINI_H743_HD/config.h:121:#define ADC3_DMA_OPT       10
src/config/configs/NEUTRONRCH7BT/config.h:121:#define ADC1_DMA_OPT        8
src/config/configs/NEUTRONRCH7BT/config.h:122:#define ADC3_DMA_OPT        9
src/config/configs/IFLIGHT_H7_TWING/config.h:103:#define ADC1_DMA_OPT        8
src/config/configs/IFLIGHT_H7_TWING/config.h:104:#define ADC3_DMA_OPT        9
src/config/configs/JHEH743_HD/config.h:123:#define ADC1_DMA_OPT 8
src/config/configs/JHEH743_HD/config.h:124:#define ADC3_DMA_OPT 9
src/config/configs/MAMBAH743/config.h:114:#define ADC1_DMA_OPT        8
src/config/configs/MAMBAH743/config.h:115:#define ADC3_DMA_OPT        9
src/config/configs/SDMODELH7V2/config.h:105:#define ADC1_DMA_OPT        8
src/config/configs/SDMODELH7V2/config.h:106:#define ADC3_DMA_OPT        9
src/config/configs/MATEKH743/config.h:138:#define ADC1_DMA_OPT        8
src/config/configs/MATEKH743/config.h:139:#define ADC3_DMA_OPT        9
src/config/configs/SPRACINGH7NANO/config.h:106:#define ADC1_DMA_OPT            8
src/config/configs/SPRACINGH7NANO/config.h:107:#define ADC3_DMA_OPT            9
src/config/configs/IFLIGHT_H743_AIO_V2/config.h:106:#define ADC1_DMA_OPT        1
src/config/configs/IFLIGHT_H743_AIO_V2/config.h:107:#define ADC2_DMA_OPT        9
src/config/configs/IFLIGHT_H743_AIO_V2/config.h:108:#define ADC3_DMA_OPT       10
src/config/configs/SPRACINGH7EXTREME/config.h:111:#define ADC1_DMA_OPT 8
src/config/configs/SPRACINGH7EXTREME/config.h:112:#define ADC3_DMA_OPT 9
src/config/configs/SPRACINGH7EXTREME/config.h:226:#define ADC1_DMA_OPT        8
src/config/configs/SPRACINGH7EXTREME/config.h:227:#define ADC3_DMA_OPT        9
src/config/configs/TMOTORH743/config.h:122:#define ADC1_DMA_OPT                    8
src/config/configs/TMOTORH743/config.h:123:#define ADC3_DMA_OPT                    9
src/config/configs/KAKUTEH7/config.h:100:#define ADC1_DMA_OPT        8
src/config/configs/KAKUTEH7/config.h:101:#define ADC3_DMA_OPT        9
src/config/configs/KAKUTEH7V2/config.h:96:#define ADC1_DMA_OPT        8
src/config/configs/KAKUTEH7V2/config.h:97:#define ADC3_DMA_OPT        9
src/config/configs/SEQUREH7/config.h:99:#define ADC1_DMA_OPT 8
src/config/configs/SEQUREH7/config.h:100:#define ADC3_DMA_OPT 9
src/config/configs/MAMBAH743_2022B/config.h:114:#define ADC1_DMA_OPT        8
src/config/configs/MAMBAH743_2022B/config.h:115:#define ADC3_DMA_OPT        9
src/config/configs/SPRACINGH7ZERO/config.h:119:#define ADC1_DMA_OPT            8
src/config/configs/SPRACINGH7ZERO/config.h:120:#define ADC3_DMA_OPT            9
src/config/configs/KAKUTEH7MINI/config.h:100:#define ADC1_DMA_OPT        8
src/config/configs/KAKUTEH7MINI/config.h:101:#define ADC3_DMA_OPT        9
src/config/configs/SPRACINGH7RF/config.h:125:#define ADC3_DMA_OPT                  10
src/config/configs/SKYSTARSH7HD/config.h:123:#define ADC1_DMA_OPT        8
src/config/configs/SKYSTARSH7HD/config.h:124:#define ADC3_DMA_OPT        9
src/config/configs/SEQUREH7V2/config.h:100:#define ADC1_DMA_OPT 8
src/config/configs/SEQUREH7V2/config.h:101:#define ADC3_DMA_OPT 9
src/config/configs/FLYWOOH743PRO/config.h:119:#define ADC1_DMA_OPT        8
src/config/configs/FLYWOOH743PRO/config.h:120:#define ADC3_DMA_OPT        9
src/config/configs/SDMODELH7/config.h:98:#define ADC1_DMA_OPT 8
src/config/configs/SDMODELH7/config.h:99:#define ADC3_DMA_OPT 9
src/config/configs/SAKURAH743/config.h:157:#define ADC1_DMA_OPT        8
src/config/configs/SAKURAH743/config.h:158:#define ADC3_DMA_OPT        9
src/config/configs/CHONKERH735/config.h:109:#define ADC1_DMA_OPT                    8
src/config/configs/CHONKERH735/config.h:110:#define ADC3_DMA_OPT                    9
src/config/configs/NEUTRONRCH743AIO/config.h:116:#define ADC1_DMA_OPT        8
src/config/configs/NEUTRONRCH743AIO/config.h:117:#define ADC3_DMA_OPT        9
src/config/configs/2RAW_THUNDER_H7/config.h:112:#define ADC1_DMA_OPT        8
src/config/configs/2RAW_THUNDER_H7/config.h:113:#define ADC3_DMA_OPT        9
src/config/configs/TMH7/config.h:107:#define ADC1_DMA_OPT        8
src/config/configs/TMH7/config.h:108:#define ADC3_DMA_OPT        9
src/config/configs/IFLIGHT_BLITZ_H7_PRO/config.h:121:#define ADC1_DMA_OPT        8
src/config/configs/IFLIGHT_BLITZ_H7_PRO/config.h:122:#define ADC3_DMA_OPT        9
src/config/configs/BETAFPVH743/config.h:111:#define ADC1_DMA_OPT        1
src/config/configs/BETAFPVH743/config.h:112:#define ADC2_DMA_OPT        9
src/config/configs/BETAFPVH743/config.h:113:#define ADC3_DMA_OPT       10
src/config/configs/JHEH743_AIO/config.h:107:#define ADC1_DMA_OPT        8
src/config/configs/JHEH743_AIO/config.h:108:#define ADC3_DMA_OPT        9
src/config/configs/FOXEERH743/config.h:105:#define ADC1_DMA_OPT                    8
src/config/configs/FOXEERH743/config.h:106:#define ADC2_DMA_OPT                    9
src/config/configs/FOXEERH743/config.h:107:#define ADC3_DMA_OPT                    10
src/config/configs/AOCODARCH7DUAL/config.h:122:#define ADC1_DMA_OPT        8
src/config/configs/AOCODARCH7DUAL/config.h:123:#define ADC3_DMA_OPT        9
```

Reply:
Wonder if `IFLIGHT_H743_AIO_V2` and `BETAFPVH743` seems to be wrong here.

Reply:
Doesn’t answer my question.. with the dma controller it’s better to let it assign dma channels